scripts/core/preflight-check.sh

```
@@ -239,28 +247,39 @@ fi
+    # the cap (matrix builds, reruns) — this repo runs ~2 workflows per
+    # push, so 50 is generous headroom. The event filter happens BELOW,
+    # not in the gh --jq, so the truncation guard sees the RAW returned
+    # count — a cap-full response of mixed events must still trip the
+    # guard (KIT-0043 F1).
     # shellcheck disable=SC2086
-    CI_RUNS=$(gh $GH_REPO_ARG run list --commit "$LATEST_SHA" --limit 10 --json status,conclusion,workflowName,event,headSha \
-        --jq '[.[] | select(.event == "push" or .event == "pull_request")]' 2>/dev/null)
+    CI_RUNS=$(gh $GH_REPO_ARG run list --commit "$LATEST_SHA" --limit "$CI_RUN_LIMIT" --json status,conclusion,workflowName,event,headSha 2>/dev/null)
     # Distinguish "gh succeeded with an empty list" (runs not registered
     # yet → PENDING) from "gh itself failed" (auth/network → FAIL below)
     if [ $? -eq 0 ]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.

(SC2181)